systemtests/bin/upload_symbols.py

```diff
@@ -39,6 +39,10 @@ def emit(self, record):
 METRICS = markus.get_metrics()
 
 
+def is_try(symbols_file):
+    return symbols_file.endswith("__try")
```

Comment:
Pretty sure this doesn't work because `symbols_file` here is something like `./data/zip-files/symbols_20240229_195629__try.zip`.

Reply:
You're correct. Unfortunately this fix alone isn't uploading the file to the try bucket in `local` at least -- so I'm looking to see how the request is handled by the Tecken upload API. Maybe I'm not passing in the right flag or maybe I need to create a separate try bucket first as part of test setup...

Reply:
Just as a thing, we often say "regular bucket" and "try bucket", but the way it's implemented is that there's a single symbols bucket and "regular symbols" get stored relative to the root of the bucket in `v1/` and "try symbols" get stored relative to the root of the same bucket in `v1/try/`. So local dev and other set up only needs to create that one symbols bucket.

Reply:
Oh, I know what's going on. "try" isn't something you can set in the upload request. Whether or not an upload goes into the regular or try bucket is dependent on the API token permissions.

For uploading to regular, the API token needs to have "UPLOAD_SYMBOLS" permission.

For uploading to try, the API token needs to have "UPLOAD_TRY_SYMBOLS" permission.

A token can't have both permissions. This is how Tecken determines where the symbols get stored.

It's a terrible system and I hate it. Fixing it is covered in https://bugzilla.mozilla.org/show_bug.cgi?id=1759740 though that also talks about private symbols which are no longer a thing.

So... what we need to do is use a different API token to upload the try symbols.

Reply:
Oh okay thanks! Hmm yes that is confusing. I will work on the fix.

Reply:
Almost done with the changes -- one thing I wanted to check:

The [bug you linked says](https://bugzilla.mozilla.org/show_bug.cgi?id=1759740#c0):
> Additionally, you can specify which bucket you're uploading to in the upload POST data. However, it looks like this is only usable by superusers of which there aren't any that would use this feature.

Since I'm a superuser (and all of us are on the Obs team), doesn't this mean I can specify the bucket with this approach?

Reply:
Interesting! We are superusers, but it doesn't seem to be working that way now. Maybe I was wrong back then? Maybe I was right, but things have changed? Maybe I broke something when I removed all the private symbols stuff mid-2023?

Reply:
Okay I think I have this working now. Just need to update some docs.

One more thing I wanted to follow up on:
> "regular symbols" get stored relative to the root of the bucket in v1/ and "try symbols" get stored relative to the root of the same bucket in v1/try/

When I use the S3 CLI locally after running the upload tests (now correctly uploading to the try bucket as specified AFAICT), the path looks to be `try/v1`, not `v1/try` -- does that make any difference?

```
app@fbd1c8fc2ee6:/app$ ./bin/s3_cli.py list_objects publicbucket
try/v1/libxul.so/D5A5A0F062BEAE76231D6CD483363C790/libxul.so.sym	108263476	2024-03-01 19:31:10+00:00
v1/BadCertAndPinningServer/88721AAAC423C52C22A84FDCE3D056E90/BadCertAndPinningServer.sym	472460	2024-03-01 19:41:45+00:00
...
```